packages/code/src/Code.tsx

```diff
@@ -140,8 +224,34 @@ function Code({
   chromeTitle = '',
   ...rest
 }: CodeProps) {
+  const [copied, setCopied] = useState(false);
+  const [canScrollLeft, setCanScrollLeft] = useState(false);
+  const [canScrollRight, setCanScrollRight] = useState(false);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const scrollableMultiLine = useRef<HTMLPreElement>(null);
+  const scrollableSingleLine = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    if (multiline) {
+      if (!isLoaded && scrollableMultiLine.current !== null) {
+        setIsLoaded(true);
```

Comment:
When you see yourself setting multiple states at once - that due to one condition two state variables need to change - it's generally better to reach for `useReducer`. [Shameless self promotion, but also more rich explanation of this pattern as a blog post I wrote.](https://hswolff.com/blog/why-i-love-usereducer/)

Reply:
My understanding of useLayoutEffect may be wrong, but could we remove this all together once useLayoutEffect replaces useEffect, because we'll have information about the layout changing rather than having to force the rerender?